hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/fileContext/ITestS3AFileContextStatistics.java

```diff
@@ -62,10 +67,25 @@ protected void verifyReadBytes(FileSystem.Statistics stats) {
     Assert.assertEquals(2 * blockSize, stats.getBytesRead());
   }
 
+  /**
+   * A method to verify the bytes written.
+   *
+   * NOTE: if Client side encryption is enabled, expected bytes written
+   * should increase by 16(padding of data) + 130(KMS key generation) in case
+   * of storage type{@link CryptoStorageMode} as ObjectMetadata(Default). If
+   * Crypto Storage mode is instruction file then add additional bytes as
+   * that file is stored separately and would account for bytes written.
+   *
+   * @param stats Filesystem statistics.
+   */
   @Override
   protected void verifyWrittenBytes(FileSystem.Statistics stats) {
     //No extra bytes are written
-    Assert.assertEquals(blockSize, stats.getBytesWritten());
+    long expectedBlockSize = blockSize;
+    if(conf.get(CLIENT_SIDE_ENCRYPTION_METHOD) != null) {
+      expectedBlockSize += 16 + 130;
```

Comment:
is that 16 some constant somewhere? If so, please switch to make it more obvious